pages/resetpassword.tsx

```diff
@@ -0,0 +1,89 @@
+import { useRouter } from "next/router";
+import React, { useState, useContext } from "react";
+import { AuthContext } from "../context/AuthContext";
+import styles from "../styles/ForgotPassword.module.css";
+import TextField from "@mui/material/TextField";
+
+const cssTextField = {
+  color: "black",
+  width: 690,
+  height: 80,
+  "& .MuiFilledInput-input": {
+    color: "#000000",
+  },
+  "& .MuiFormLabel-root": {
+    color: "#494C4E",
+  },
+};
+
+const ResetPassword: React.FC = () => {
+  const auth = useContext(AuthContext);
+  const router = useRouter();
+
+  const [newPassword, setNewPassword] = useState<string>("");
+  const [confirmNewPassword, setConfirmNewPassword] = useState<string>("");
+  const [error, setError] = useState<Error | null>(null);
+
+  // firebase requires passwords to be greater than 6 characters
+  const passwordLengthOk = newPassword == "" || newPassword.length >= 6;
+  const passwordsMatch = newPassword == "" || newPassword === confirmNewPassword;
+
+  const errorMessage =
+    error !== null
+      ? error.message
+      : !passwordLengthOk
+      ? "Passwords must be greater than 6 characters"
+      : !passwordsMatch
+      ? "Passwords do not match"
+      : null;
+
+  const handleResetPassword = async (): Promise<void> => {
+    //retrieve code from url
+    const queryParams = new URLSearchParams(window.location.search);
+    const code = queryParams.get("oobCode")?.toString();
+    if (passwordLengthOk && passwordsMatch && code) {
```

Comment:
instead of checking this way, we need to disable the button so that a user cannot click it if there is an error being displayed. Just decalare a `disabled` variable that does the following `const notDisabled = passwordLengthOk && passwordsMatch` and then pass `!notDisabled` into the disabled parameter for the button